Created company module

43r63w opened the pull request.

43r63w committed:
Created company module

Puchkof commented on Insurer/Insurer.Host/Configuration/GlobalExceptionHandler.cs:
```diff
@@ -0,0 +1,25 @@
+    {
+        var problemDetails = new ProblemDetails
+        {
+            Status = StatusCodes.Status500InternalServerError,
```
Maybe we should handle different status codes?

Puchkof commented on Insurer/Modules/Company/Company.Domain/Entity/Document.cs:
```diff
@@ -0,0 +1,23 @@
+
+    public FileType Type { get; set; }
+
+    public double Size { get; set; }
```
Do you need to store size? Also do you have limits in app layer?

43r63w committed:
changed route for endpoits

43r63w committed:
Created file storage repository and document service

Puchkof commented on Insurer/Modules/Company/Company.Application/Services/DocumentService.cs:
```diff
@@ -18,13 +22,32 @@ public async Task<Result<bool>> CreateDocumentAsync(CreateDocumentModel model,
+
+        var createFileModel = new CreateFileModel
+        {
+            ObjectKey = $"/documents/{Guid.NewGuid()}/{model.Name}",
```
why the guid is before the name? is it needed?

Puchkof commented on Insurer/Modules/Company/Company.Application/Models/CreateFileModel.cs:
```diff
@@ -0,0 +1,14 @@
+
+    public string Type { get; set; } = null!;
+
+    public string? CompanyName { get; set; } = null!;
```
Maybe use companyId?

Puchkof commented on Insurer/Modules/Company/Company.Application/Models/CreateFileModel.cs:
```diff
@@ -0,0 +1,14 @@
+
+    public string? CompanyName { get; set; } = null!;
+
+    public string? InsuranceType { get; set; }
```
enum?

Puchkof commented on Insurer/Modules/Company/Company.Infrastructure/FileStorage/MinioSettings.cs:
```diff
@@ -0,0 +1,18 @@
+
+    public bool UseSsl { get; set; }
+
+    public int FileSize { get; set; }
```
MaxFileSize

43r63w committed:
Removed public service and added switch statment in global exception handler

43r63w committed:
Changed all responses from services

43r63w committed:
Add document management feature and antiforgery

Introduced new document management endpoints for creating, retrieving, and deleting documents. Refactored `DocumentService` and related models to support these operations. Updated the database schema to include a `UserId` in the `Document` entity and added migrations for schema changes. Added antiforgery services and middleware to enhance security. Created a new project for the document domain targeting .NET 9.0.

43r63w committed:
Completed company module

MuzykaD commented on Insurer/Modules/Document/Document.Domain/Class1.cs:
```diff
@@ -0,0 +1,6 @@
+﻿namespace Document.Domain;
+
+public class Class1
```
Delete it

MuzykaD commented on Insurer/Modules/Company/Company.Application/Company.Application.csproj:
```diff
@@ -16,4 +16,10 @@
 
+    <ItemGroup>
+      <Reference Include="Microsoft.AspNetCore.Http.Features">
+        <HintPath>C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\9.0.9\Microsoft.AspNetCore.Http.Features.dll</HintPath>
```
What is this?

MuzykaD replied:
if you really need it - use relative path

MuzykaD commented on Insurer/Modules/Company/Company.Application/Services/DocumentService.cs:
```diff
@@ -1,54 +1,56 @@
-        var company = await companyDbContext.Companies
-            .Select(e => new { e.Id, e.CompanyName })
-            .FirstOrDefaultAsync(e => e.Id == model.CompanyId, cancellationToken);
+        var fileSize = Math.Round(file.Length / (1024.0 * 1024.0), 2);
```
Name this function somehow and use constants instead of magic numbers.
It has to be clear what this code does

MuzykaD commented on Insurer/Modules/Company/Company.Application/Services/DocumentService.cs:
```diff
@@ -1,54 +1,56 @@
+
+    public async Task<Result<CreateDocumentResponse>> CreateDocumentAsync(
+        IFormFile file,
+        string userId,
```
Do not pass user id as a param.
Pass Scoped service instead and fetch it here.

MuzykaD commented on Insurer/Modules/Company/Company.Application/Validators/CreateDocumentValidator.cs:
```diff
@@ -5,17 +5,17 @@ namespace Company.Application.Validators;
-            .NotNull()
-            .WithMessage("Type is required.");
-    }
+    // public CreateDocumentValidator()
```
Delete it

MuzykaD commented on Insurer/Insurer.Host/Endpoints/DocumentEndpoints.cs:
```diff
@@ -0,0 +1,62 @@
+        var group = app.MapGroup("documents");
+
+
+        group.MapPost("", CreateDocumentAsync).DisableAntiforgery();
```
routes to constants
